plugin/trino-hive/src/main/java/io/trino/plugin/hive/util/HiveBucketing.java

```diff
@@ -76,9 +77,9 @@ int getBucketHashCode(List<TypeInfo> types, Object[] values)
             }
 
             @Override
-            int getBucketHashCode(List<TypeInfo> types, Page page, int position)
+            int getBucketHashCode(List<TypeInfo> types, Page page, int prefixChannels, int position)
```

Comment:
I don't like this. I think it is caller responsibility to provide a page which matches the types. It is not huge cost to create page from a subset of channels before passing it to `getBucketHashCode`

Reply:
> prefixChannels is not needed too. It can be derived from types.size()
True, but that would make the API the same as the existing one, which would be confusing and we wouldn't be able to do validation that `types.size() == page.getChannelCount()` for existing usage.

>  I think it is caller responsibility to provide a page which matches the types. It is not huge cost to create page from a subset of channels before passing it to getBucketHashCode
The problem with that is the calling function `int getBucket(Page page, int position)` would have to create page from subset of channels for every row. This seems wasteful even if we are not copying the blocks.

Reply:
> > I think it is caller responsibility to provide a page which matches the types. It is not huge cost to create page
> > from a subset of channels before passing it to getBucketHashCode
> 
> The problem with that is the calling function `int getBucket(Page page, int position)` would have to create page from subset of channels for every row. This seems wasteful even if we are not copying the blocks.

Yeah - you are right. Keep it this way then.